transport/http/src/main/java/org/kaazing/gateway/transport/http/HttpAcceptor.java

```diff
@@ -467,6 +470,37 @@ public DefaultHttpSession call() {
                         SESSION_KEY.set(session, newHttpSession);
                         return newHttpSession;
                     }
+
+
+
+                    private IoHandler getIoHandler(DefaultHttpSession newHttpSession, boolean invalidHeaders) {
```

Comment:
Hi David,

I would agree if the code would have been written differently.

However, the reason why I chose this particular place is because you need access to the session in order to be able to send back the proper HTTP 400 response. Placing it in the HttpAcceptor is the easier way of doing it. 

The issue is, as ever, the Pokemon exceptions. I would have to fix them everywhere in this code and this bug would take quite a while to fix. 

Let me explain it more clearly:

Within HttpHeaderDecoding we would have to throw a ProtocolDecoderException exception. Now the code, is done in such a way that the function that actually interprets this exception closes the connection. This is not what we want (we want to send back HTTP 400).

The way to do it would be to interpret the exception, and if it is a protocol decoding error we would then send back HTTP 400. This is not doable because the function calling the decoding state machine wraps all the exceptions as "Exception". Anyway, the same old anti-pattern...

Anyway, if you want it done in that particular way, I can do it (after all, that is the proper way of doing it), but I suggest a thorough refactoring if we are going to do it in that particular way.

Kindest regards,
Doru